client/state/selectors/is-domain-sidebar-experiment-user.js

```diff
@@ -1,15 +0,0 @@
-import { getCurrentUser } from 'calypso/state/current-user/selectors';
-import getCurrentQueryArguments from 'calypso/state/selectors/get-current-query-arguments';
-
-/**
- * Whether the user is in the Domain sidebar upsell experiment.
- *
- * @param state {object} Global state tree.
- * @returns {boolean} Whether the user is in the Domain sidebar upsell experiment.
- */
-export const isDomainSidebarExperimentUser = ( state ) => {
-	const currentUser = getCurrentUser( state );
-	const domainAndPlanPackage = !! getCurrentQueryArguments( state )?.domainAndPlanPackage;
-
-	return domainAndPlanPackage && 'treatment' === currentUser?.calypso_sidebar_upsell_experiment;
```

Comment:
Seems like we can also remove the `calypso_sidebar_upsell_experiment` key from:

1) The list of allowed additional user attributes:

https://github.com/Automattic/wp-calypso/blob/55db65755421ed2a147412a57dc527abda66be42/client/lib/user/shared-utils/filter-user-object.js#L35

2) The optional user data that is part of the user type:

https://github.com/Automattic/wp-calypso/blob/55db65755421ed2a147412a57dc527abda66be42/client/lib/user/user.d.ts#L47

Reply:
Thanks for your suggestion, I totally missed those 😞